src/EnumDisplayName/EnumExtensions.cs

```diff
@@ -35,26 +36,87 @@ public static string GetDisplayName(this Enum enumValue)
 
                 if (string.Compare(fieldName, field.Name, StringComparison.Ordinal) == 0)
                 {
-                    var customAttributes = field.GetCustomAttributes(false);
-                    var enumDisplayNameAttribute = customAttributes.OfType<EnumDisplayNameAttribute>().FirstOrDefault();
+                    var fieldAttributes = field
+                        .GetCustomAttributes(false)
+                        .Cast<Attribute>();
 
-                    if (enumDisplayNameAttribute != null)
-                    {
-                        return enumDisplayNameAttribute.GetDisplayName();
-                    }
+                    return GetNameFromEnumField(enumValue, fieldAttributes);
+                }
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Returns the display names of a flag enum as a comma seperated string as specified by the
+        /// EnumDisplayNameAttribute declared on the enum field. If no EnumDisplayNameAttribute is
+        /// declared on the enum field then the name of the enum field is returned.
+        /// </summary>
+        public static string GetFlagsDisplayName(this Enum enumValue)
+        {
+            if (enumValue == null)
+            {
+                return string.Empty;
+            }
 
-                    var displayNameAttribute = customAttributes.OfType<DisplayAttribute>().FirstOrDefault();
+            var enumType = enumValue.GetType();
+            var fieldNames = Enum.GetNames(enumType);
+            var fields = enumType.GetFields()
+                .Where(o => fieldNames.Any(p => p == o.Name))
```

Comment:
Consider using `Contains` and specifying string comparison typeof `Ordinal` (_not_ ignore case)